src/posit/connect/resources.py

```diff
@@ -50,64 +51,61 @@ def __init__(self, params: ResourceParameters) -> None:
 
         Parameters
         ----------
         ctx : Context
             The context object containing the session and URL for API interactions.
-        parent : Optional[Active], optional
-            An optional parent resource that establishes a hierarchical relationship, by default None.
-        **kwargs : dict
-            Additional keyword arguments passed to the parent `Resource` class.
+        path : str
+            The HTTP path component for the collection endpoint
+        pathinfo : str
+            The HTTP part of the path directed at a specific resource
+        **attributes : dict
+            Resource attributes passed
+
+        Attributes
+        ----------
+        _ctx : Context
+            The context object containing the session and URL for API interactions
+        _path : str
+            The HTTP path for the collection endpoint.
         """
         params = ResourceParameters(ctx.session, ctx.url)
-        super().__init__(params, **kwargs)
+        super().__init__(params, **attributes)
         self._ctx = ctx
-        self._parent = parent
+        self._path = posixpath.join(path, pathinfo)
 
 
 T = TypeVar("T", bound="Active")
 """A type variable that is bound to the `Active` class"""
 
 
 class ActiveSequence(ABC, Generic[T], Sequence[T]):
-    def __init__(self, ctx: Context, parent: Optional[Active] = None):
+    def __init__(self, ctx: Context, path: str, pathinfo: str = "", uid: str = "guid"):
         """A sequence abstraction for any HTTP GET endpoint that returns a collection.
 
         It lazily fetches data on demand, caches the results, and allows for standard sequence operations like indexing and slicing.
 
-        Parameters
+        Attributes
         ----------
-        ctx : Context
-            The context object that holds the HTTP session used for sending the GET request.
-        parent : Optional[Active], optional
-            An optional parent resource to establish a nested relationship, by default None.
+        _ctx : Context
+            The context object containing the session and URL for API interactions
+        _path : str
+            The HTTP path for the collection endpoint.
+        _uid : str
+            The field name used to uniquely identify records.
+        _cache: Optional[List[T]]
         """
         super().__init__()
         self._ctx = ctx
-        self._parent = parent
+        self._path = posixpath.join(path, pathinfo)
+        self._uid = uid
         self._cache: Optional[List[T]] = None
 
-    @property
-    @abstractmethod
-    def _endpoint(self) -> str:
-        """
-        Abstract property to define the endpoint URL for the GET request.
-
-        Subclasses must implement this property to return the API endpoint URL that will
-        be queried to fetch the data.
-
-        Returns
-        -------
-        str
-            The API endpoint URL.
-        """
-        raise NotImplementedError()
-
     @property
```

Comment:
This method performs a lot of side effects. I would suggest dropping `@property` to help signal things are being performed.

```suggestion
```

Reply:
Good call! It turns out I've been misusing the property annotation 🙃